core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -562,11 +573,11 @@ def _initialize(self, X=None, y=None, feature_metadata=None, num_classes=None, *
 
         self._init_misc(X=X, y=y, feature_metadata=feature_metadata, num_classes=num_classes, **kwargs)
 
+        self._init_params()
```

Comment:
qq: why did we move it here?

Reply:
Good question!

- In `self._preprocess_set_features` I added logic in this PR which updates `self.features to reflect which base_models are being used. Previously `self.features` wasn't correct and contained unused base models.
- In order to know which `base_models` are being used, I need to run the logic which filters base_models based on `max_models` and `max_models_per_type`.
- In order to run the filtering logic, I first need to know the hyperparameter values of `max_models` and `max_models_per_type`.
- In order to know those hyperparameter values, I need the hyperparameters initialized.
- `self._init_params()` initializes the parameters so I can use them in `self._preprocess_set_features`.